rdmd.d

```diff
@@ -329,12 +329,14 @@ int main(string[] args)
 
 size_t indexOfProgram(string[] args)
 {
-    foreach(i, arg; args[1 .. $])
+    foreach(i; 1 .. args.length)
     {
+        auto arg = args[i];
         if (!arg.startsWith('-', '@') &&
-                !arg.endsWith(".obj", ".o", ".lib", ".a", ".def", ".map", ".res"))
+                !arg.endsWith(".obj", ".o", ".lib", ".a", ".def", ".map", ".res") &&
+                args[i - 1] != "--eval")
```

Comment:
@Superstar64 FYI: I fixed the style here. DStyle is to use whitespace before/after binary operators.